java/README.md

```diff
@@ -0,0 +1,142 @@
+
+### To run locally
+
+```sh
+
+```
+
+
+## Structure
+
+
+### pom.xml
+https://github.com/digital-asset/hub-pythonbot-example/blob/51b05c41849dc98b821cc3a1244b0235f3bc0339/java/pom.xml#L1-L89
+
+### Directory structure
+```
+├── Makefile
+├── README.md
+├── log
+│   ├── canton.log
+│   └── canton_errors.log
+├── pom.xml
+├── run_local.sh
+├── src
+│   └── main
+│       ├── java
+│       │   └── examples
+│       │       └── javabot
+│       │           ├── Main.java
+│       │           ├── Processor.java
+│       │           └── codegen
+│       │               ├── da
+│       │               │   ├── internal
+│       │               │   │   └── template
+│       │               │   │       └── Archive.java
+│       │               │   └── types
+│       │               │       └── Tuple2.java
+│       │               ├── main
+│       │               │   ├── JavabotRequest.java
+│       │               │   ├── JavabotRequest_Accept.java
+│       │               │   ├── JavabotResponse.java
+│       │               │   ├── PythonbotRequest.java
+│       │               │   ├── PythonbotResponse.java
+│       │               │   ├── Pythonbot_Accept.java
+│       │               │   ├── TriggerRequest.java
+│       │               │   ├── TriggerResponse.java
+│       │               │   └── Trigger_Accept.java
+│       │               └── user
+│       │                   ├── Acknowledge.java
+│       │                   ├── Alias.java
+│       │                   ├── Change.java
+│       │                   ├── Follow.java
+│       │                   ├── Notification.java
+│       │                   └── User.java
+│       └── resources
+│           └── logback.xml
+```
+
+### `main`
+
+```java
+public class Main {
+
+    public static void main(String[] args) {
+
+        String appId = System.getenv("DAML_APPLICATION_ID");
+        String ledgerId = System.getenv("DAML_LEDGER_ID");
+        String userId = System.getenv("DAML_USER_ID");
+        String url[] = System.getenv("DAML_LEDGER_URL").split(":");
+        String host = url[0];
+        int port = Integer.parseInt(url[1]);
+    }
+}
+```
+
+## Configuration
+
+You can configure an automation that may differ from deployment to deployment by uploading a configuration file when the automation is deployed.
+
+This configuration can be of any type, for example `.json`, `.toml`, `.yaml`.
+
+To access this configuration file from the automation, set the environment variable `CONFIG_FILE` to point to a location on the volume where the configuration file is stored. This can then be read by the automation and parsed based on what type of file was uploaded.
+
+For example if a JSON file was uploaded:
```

Comment:
I really like this example. Could we also add an example of what a config file looks like, or link to one in the repo?